app/controllers/api/base_controller.rb

```diff
@@ -55,4 +62,8 @@ def render_api_key_forbidden
       format.yaml { render yaml: { error: t(:api_key_forbidden) }, status: :forbidden }
     end
   end
+
+  def render_soft_deleted_api_key
+    render plain: "An invalid API key cannot be used. Please delete it and create a new one.", status: :forbidden
```

Comment:
Are we able to say why it's invalid?

Reply:
We could by process of elimination. Since gem ownership deletion is the only way to invalidate an api key, we know it's the reason. I don't think it's necessary to give more information, wdyt?